bootstrap/cmd/kfctl/cmd/apply.go

```diff
@@ -47,6 +47,10 @@ var applyCmd = &cobra.Command{
 		if applyErr != nil {
 			return fmt.Errorf("couldn't apply KfApp: %v", applyErr)
 		}
+		postApplyErr := kfApp.PostApply(resource)
```

Comment:
Thinking about this approach, adding a new interface method 'PostApply' and hoisting its call all the way up to the cobra cmd level seems perhaps heavy-handed. PostApply is driven by the need to call the platform to create a DefaultPod after the package manager is done? I say heavy-handed because it seems like the only implementation of PostApply is Gcp but the cost is having all implementations stub-out the method.

One suggestion would be to return a continuation from Apply shown below
```
type Afterwards func() error
type KfApp interface {
	Apply(resources ResourceEnum) (Afterwards, error)
	Delete(resources ResourceEnum) error
	Generate(resources ResourceEnum) error
	Init(resources ResourceEnum) error
}
```

and just have coordinator call this if it's not nil at the end of its Apply.
Gcp would return a continuation that looks similar to [gcpAddedConfig](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/pkg/kfapp/coordinator/coordinator.go#L764)
but much simpler and defined in Gcp.

The gcpAddedConfig in coordinator.Apply
```
	gcpAddedConfig := func() error {
		if kfapp.KfDef.Spec.Email == "" || kfapp.KfDef.Spec.Platform != kftypesv3.GCP {
			return nil
		}

		if p, ok := kfapp.Platforms[kfapp.KfDef.Spec.Platform]; !ok {
			return &kfapis.KfError{
				Code:    int(kfapis.INTERNAL_ERROR),
				Message: "Platform GCP specified but not loaded.",
			}
		} else {
			gcp := p.(*gcp.Gcp)
			p, err := gcp.GetPluginSpec()
			if err != nil {
				return err
			}
			if *p.EnableWorkloadIdentity {
				return gcp.SetupDefaultNamespaceWorkloadIdentity()
			} else {
				return gcp.ConfigPodDefault()
			}
		}
	}
```
Is moved to Gcp in its Apply method
```
	continuation := func() error {
		p, err := gcp.GetPluginSpec()
		if err != nil {
			return err
		}
		if *p.EnableWorkloadIdentity {
			return gcp.SetupDefaultNamespaceWorkloadIdentity()
		} else {
			return gcp.ConfigPodDefault()
		}
	}
        return continuation
```

So the coordinator could just call this at this end of its Apply or could maintain a stack like so:
```
  cb, err := platform.Apply()
  if err != nil ...
  if cb != nil coordinator.push(cb)

  cb, err := packagemanager.Apply()
  if err != nil ...
  if cb != nil coordinator.push(cb)

  while cb = coordinator.pop(); cb != nil {
    cb(resource)
  }
```

Reply:
Thanks for raising this @kkasravi ; I didn't realize we were exposing this as a new top level command.

@yanniszark @gabrielwen I'd really prefer not to introduce any more user visible changes (e.g. new command); so if we do need to go with a PostApply hook lets implement it so as not be user visible.